src/fondue/Fondue.js

```diff
@@ -798,3 +942,76 @@ export default class Fondue {
 		return allGlyphs;
 	}
 }
+
+const createType6Summary = (feature, randomize) => {
+	let allInputs = [];
+	let allBacktracks = [];
+	let allLookaheads = [];
+
+	const limit = 10; // Summary will be limited to a max of limit³ (e.g. 20*20*20 = 8000)
+
+	const shuffleArray = (array) => {
+		for (let i = array.length - 1; i > 0; i--) {
+			const j = Math.floor(Math.random() * (i + 1));
+			[array[i], array[j]] = [array[j], array[i]];
+		}
+		return array;
+	};
+
+	// Create some kind of "all backtracks" or "all lookaheads"
+	for (const lookup of feature["lookups"]) {
+		if (lookup.type !== 6) continue;
+
+		// Create all possible combinations of input, backtrack and lookahead
+		for (const key in Object.entries(lookup["input"])) {
+			allInputs = [...new Set(allInputs.concat(lookup["input"][key]))];
+
+			if (lookup["backtrack"][key]) {
+				allBacktracks = [
+					...new Set(allBacktracks.concat(lookup["backtrack"][key])),
+				];
+			}
+
+			if (lookup["lookahead"][key]) {
+				allLookaheads = [
+					...new Set(allLookaheads.concat(lookup["lookahead"][key])),
+				];
+			}
+		}
+	}
+
+	if (randomize) {
+		allInputs = shuffleArray(allInputs);
+
+		if (allBacktracks.length) {
+			allBacktracks = shuffleArray(allBacktracks);
+		}
+
+		if (allLookaheads.length) {
+			allLookaheads = shuffleArray(allLookaheads);
+		}
+	}
+
+	let allCombinations = [allInputs.slice(0, limit)];
+
+	if (allBacktracks.length) {
+		allCombinations.unshift(allBacktracks.slice(0, limit));
+	}
+
+	if (allLookaheads.length) {
+		allCombinations.push(allLookaheads.slice(0, limit));
+	}
+
+	let summarizedCombinations = allCombinations
+		.reduce((a, b) =>
+			a.reduce((r, v) => r.concat(b.map((w) => [].concat(v, w))), [])
+		)
+		.map((a) => a.join(""));
```

Comment:
It's not really clear to me what this does. Can you either clarify the code, or add some comments?